lms/djangoapps/grades/models.py

```diff
@@ -376,6 +423,14 @@ def _prepare_params_visible_blocks_id(cls, params):
         params['visible_blocks_id'] = params['visible_blocks'].hash_value
         del params['visible_blocks']
 
+    def remove_attempts(self):
```

Comment:
What is the use case for this method? I thought that once a subsection was attempted, it was attempted for good.

If that's not the case, this should get called in [reset_student_attempts](https://github.com/edx/edx-platform/blob/bc551b9b2566c8552f057e8bc4064beac2722752/lms/djangoapps/instructor/enrollment.py#L204) if delete_module is true.